roxie/ccd/ccdstate.cpp

```diff
@@ -724,7 +724,7 @@ class CRoxiePackageNode : extends CPackageNode, implements IRoxiePackage
         Owned<IRoxieDaliHelper> daliHelper = connectToDali();
         bool onlyLocal = fileNameServiceDali.isEmpty();
         bool onlyDFS = !resolveLocally() && !onlyLocal;
-        Owned<ILocalOrDistributedFile> ldFile = createLocalOrDistributedFile(fileName, NULL, onlyLocal, onlyDFS, true);
+        Owned<ILocalOrDistributedFile> ldFile = createLocalOrDistributedFile(fileName, wu->queryUserDescriptor(), onlyLocal, onlyDFS, true);
```

Comment:
Not sure you can assume wu is not null.

In adhoc mode you want to use userdescriptor from wu, in predeployed query mode you need to use the global roxie user info.